### PR TITLE
DB: Implemented schema for database

### PR DESCRIPTION
**Pull request by MuskanPaliwal**
Have not tested the schema yet as I wanted to be sure with the fields we are adding in the tables. Please let me know if there are some fields left out in the tables.

### CODE REVIEW

File: facultyRecruitment/jobs/models.py [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
 from django.db import models
+from django.db.models.base import Model
+from django.db.models.deletion import CASCADE
+from facultyRecruitment.facultyRecruitment.settings import DATE_INPUT_FORMATS
+from django.utils import timezone
+from django.contrib.auth.models import User
 
-# Create your models here.
+class Jobs(models.Model):
+    jobId = models.CharField(primary_key=True, max_length=10)
```
Comment (amitsaini4556) [inline facultyRecruitment/jobs/models.py:10]: I think jobId is an integer type, not a character? 
or replace Chatfield with AutoField and remove max_length constraint.

File: facultyRecruitment/jobs/models.py [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+    adminId = models.ForeignKey(User, on_delete=models.CASCADE)
+    title = models.CharField(max_length=50)
+    position = models.CharField(max_length=40)
+    applyBy = models.DateField(input_formats=DATE_INPUT_FORMATS)
+    datePosted = models.DateTimeField(default=timezone.now)
+    description = models.TextField()
+    collegeName = models.CharField(max_length=50)
+    availablePositions = models.IntegerField()
+
```
Comment (amitsaini4556) [inline facultyRecruitment/jobs/models.py:19]: ImageField missing for the image.

File: facultyRecruitment/jobs/models.py [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+    ('inProgress', _('Application is in progress')),
+    ('accepted', _('Accepted by admin to proceed for further round of interviews')),
+    ('rejected', _('Rejected - does not fit well in the job profile, hence rejected')),
+    )
+
+    userId = models.ForeignKey(User, on_delete=models.CASCADE, unique=True)
+    jobId = models.ForeignKey(Jobs, on_delete=models.CASCADE, unique=True)
+    status = models.CharField(max_length=50, choices=STATUS, default='inProgress')
+    name = models.CharField(max_length=40)
+    email= models.CharField(max_length=40)
```
Comment (amitsaini4556) [inline facultyRecruitment/jobs/models.py:34]: use EmailField for the email field

File: facultyRecruitment/jobs/models.py [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+    phoneNumber = models.CharField(max_length=12)
```
Comment (amitsaini4556) [inline facultyRecruitment/jobs/models.py:35]: use PhoneField

**Review by amitsaini4556 — CHANGES_REQUESTED** [review]

File: facultyRecruitment/jobs/models.py [hunk 5]
```diff
@@ -1,3 +1,44 @@
...
-# Create your models here.
+class Jobs(models.Model):
+    jobId = models.CharField(primary_key=True, max_length=10)
+    adminId = models.ForeignKey(User, on_delete=models.CASCADE)
+    title = models.CharField(max_length=50)
+    position = models.CharField(max_length=40)
+    applyBy = models.DateField(input_formats=DATE_INPUT_FORMATS)
+    datePosted = models.DateTimeField(default=timezone.now)
+    description = models.TextField()
+    collegeName = models.CharField(max_length=50)
```
Comment (hemabhagnani) [inline facultyRecruitment/jobs/models.py:17]: As we have limited values shouldn't we use choices in `collegeName` ?

**Review by hemabhagnani — CHANGES_REQUESTED** [review]

File: facultyRecruitment/jobs/models.py (hunk 1 above)
Comment (MuskanPaliwal) [reply facultyRecruitment/jobs/models.py:10]: Alright.

File: facultyRecruitment/jobs/models.py [hunk 6]
```diff
@@ -1,3 +1,44 @@
...
+
+class Applicants(models.Model):
+    STATUS = (
+    ('inProgress', _('Application is in progress')),
+    ('accepted', _('Accepted by admin to proceed for further round of interviews')),
+    ('rejected', _('Rejected - does not fit well in the job profile, hence rejected')),
+    )
+
+    userId = models.ForeignKey(User, on_delete=models.CASCADE, unique=True)
+    jobId = models.ForeignKey(Jobs, on_delete=models.CASCADE, unique=True)
```
Comment (hemabhagnani) [inline facultyRecruitment/jobs/models.py:31]: `userId` and `jobId` must be used as a composite primary key

File: facultyRecruitment/jobs/models.py (hunk 3 above)
Comment (MuskanPaliwal) [reply facultyRecruitment/jobs/models.py]: applied the changes.

File: facultyRecruitment/jobs/models.py [hunk 7]
```diff
@@ -1,3 +1,44 @@
...
+    status = models.CharField(max_length=50, choices=STATUS, default='inProgress')
+    name = models.CharField(max_length=40)
+    email= models.CharField(max_length=40)
+    phoneNumber = models.CharField(max_length=12)
+    location = models.TextField(max_length=100)
+    resume = models.URLField(max_length=200)
+    cv = models.URLField(max_length=200)
```
Comment (amitsaini4556) [inline facultyRecruitment/jobs/models.py:38]: Applicants fill resume or cv as a link(drive link) or file(Pdf)?

**Review by hemabhagnani — APPROVED** [review]

File: facultyRecruitment/jobs/models.py (hunk 7 above)
Comment (hemabhagnani) [reply facultyRecruitment/jobs/models.py:38]: Yes, it will be drive links for now, later will work on incorporating pdfs as well.

Comment (amitsaini4556) [reply facultyRecruitment/jobs/models.py:38]: yet not ? I think choose a file is easy rather than copy or paste the link of drive.

Comment (amitsaini4556) [reply facultyRecruitment/jobs/models.py:38]: Replace `urlField` by `FileField` for both resume or cv.

Comment (MuskanPaliwal) [reply facultyRecruitment/jobs/models.py:38]: [saving-files-in-database](https://stackoverflow.com/questions/38628770/how-to-save-files-to-database-in-django), the answer to the question clearly suggests that it is a bad idea to store files in the DB instead we could use external storage for storing files. Here are a bunch of nice options to go for [Django-Storage-Options](https://django-storages.readthedocs.io/en/latest/).

Comment (amitsaini4556) [reply facultyRecruitment/jobs/models.py:38]: ok, go for it.

**Review by amitsaini4556 — APPROVED** [review]

**Pull request merged**